webknossos/examples/accessing_metadata.py

```diff
@@ -0,0 +1,32 @@
+import webknossos as wk
+from webknossos.client.api_client.models import ApiMetadata
```

Comment:
The `api_client.models` classes should stay private to our own code, to enable future changes to the HTTP API without changing the user-exposed python API.

Please create an abstraction layer between ApiMetadata and the user-exposed metadata in Dataset and Folder classes. Maybe even without a `type` property, but using python types instead? The type property could be inferred in the conversion. 
Maybe it could even be just a dict? Not sure about that, maybe @philippotto or @MichaelBuessemeyer have a preference.

Reply:
I think, a dict mapping from key to primitive would be cool, but I'm not sure it's worth the complexity. in the end, this api should be easy and clear to use. directly using primitives might make people believe that they can also put python objects there. so, it would definitely need thorough validation.
I don't have a strong opinion about this, though.

Reply:
> because it seems that the key is not unique so far

to which implementation are you referring? the wklibs one? because in wk, it should be unique (if not, we should fix it :sweat_smile:)

Reply:
The wk one, it seems that the creation of duplicates is restricted by the frontend but not in the backend. I can send the same key-value pair multiple times with the API without an error. I just tested it with the `l4_test_mark` dataset.
![Screenshot 2024-09-05 at 09-58-33 WEBKNOSSOS](https://github.com/user-attachments/assets/3f628016-9053-4dcd-a7a0-dd75d4c9bdf2)

@MichaelBuessemeyer

Reply:
Sorry for entering the thread so late
>because in wk, it should be unique (if not, we should fix it 😅)

The frontend does the validation that no key is duplicated. The backend just takes what the frontend sends and saves & serves it. 

Do you think a validation would be helpful? I can certainly build that :)

Reply:
> Do you think a validation would be helpful?

I changed the wk-libs implementation to use a dict now. So when using the libs as intended, the key should be unique too. Nevertheless, a validation would be helpful to ensure that the API is used correct.

Reply:
> Do you think a validation would be helpful? I can certainly build that :)

We should either fully support duplicates, or ensure that none arise. So with the current design, I’d say yes, the backend should assert this in the update routes. Could you create an issue for that? It’s not super urgent, though.

Reply:
I created an issue for this: https://github.com/scalableminds/webknossos/issues/8068

And as I wanted a rather quick and simple task right now, I simply started the implementation :see_no_evil: